modules/virt-infer-instancetype-preference.adoc

```diff
@@ -0,0 +1,28 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/virt-specifying-instance-preference.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="virt-infer-instancetype-preference_{context}"]
+= Inferring an instance type or preference
+
+You can use the `--infer-instancetype` and `--infer-preference` flags to infer which instance type, preference, or both to use to define the workload sizing and runtime characteristics of a VM.
+
+Inferring instance types, preferences, or both is enabled by default, and the`inferFromVolumeFairlure` policy of the `inferFromVolume` attribute is set to `Ignore`. Any errors when inferring from the boot volume are ignored, and the VM is created with instancetype and preference left unset.
+
+However, when flags are applied, the `inferFromVolumeFailure` policy defaults to `Reject`. Any errors when inferring from the boot volume result in the rejection of the creation of that VM.
+
+.Prerequisites
+
+* The `virtctl` tool must be installed.
+
+.Procedure
+
+To explicitly infer instance types from the volume used to boot the virtual machine, use the `--infer-instancetype` flag. To explicitly infer preferences, use the `--infer-preference` flag. The following example includes both flags:
+
++
+[source,terminal]
+----
+$ virtctl create vm --infer-instancetype --infer-preference
+----
+
```

Comment:
@0xFelix Can you answer this?

Reply:
Yes, we have it in 4.15+.

These switches allow to infer the instancetype and preference from another volume than the boot volume. You can explicitly choose which volume to infer from.

Reply:
@0xFelix Is that something we need to add to this doc?

Reply:
Yes, I think adding it makes sense.

Reply:
@0xFelix Because this wasn't in the original draft and will take additional time to develop, I'm going to leave the current content as-is and request that you create an additional Jira to document `--infer-instancetype-from $volume` and `--infer-preference-from $volume` and avoid scope creep.
@ctomasko

Reply:
I created https://issues.redhat.com/browse/CNV-48102 for it.